srcCpp/ParameterManager.h

```diff
@@ -0,0 +1,120 @@
+#ifndef __PARAMETERMANAGER_H__
+#define __PARAMETERMANAGER_H__
+
+/*
+ * (c) 2005-2018  Copyright, Real-Time Innovations, Inc. All rights reserved.
+ * Subject to Eclipse Public License v1.0; see LICENSE.md for details.
+ */
+#include "PerftestTransport.h"
+#include "RTIDDSImpl.h"
+#include "Parameter.h"
+#include <map>
+
+#if defined(RTI_WIN32)
+  #pragma warning(push)
+  #pragma warning(disable : 4996)
+  #define STRNCASECMP _strnicmp
+#elif defined(RTI_VXWORKS)
+  #define STRNCASECMP strncmp
+#else
+  #define STRNCASECMP strncasecmp
+#endif
+#define IS_OPTION(str, option) (STRNCASECMP(str, option, strlen(str)) == 0)
+
+class ParameterManager
+{
+    private:
+        std::map<std::string, AnyParameter> parameterList;
+
+    public:
+        ParameterManager();
+        void initialize();
+        static ParameterManager &GetInstance();
+        ~ParameterManager();
+
+        // Set the value of a parameter
+        template <typename T>
+        void set(std::string parameterKey, T var){
+            std::map<std::string, AnyParameter>::iterator it;
+            it = parameterList.find(parameterKey);
+            if (it != parameterList.end()) {
```

Comment:
`if (parameterList.count(parameterKey)){ .. }`

This should do the trick. [std::count](http://www.cplusplus.com/reference/map/map/count/)

Reply:
Done using find

```
if (parameterList.find(parameterKey) != parameterList.end()) {
```